client/pk-console.c

```diff
@@ -833,6 +824,46 @@ pk_console_finished_cb (GObject *object, GAsyncResult *res, gpointer data)
 	g_main_loop_quit (ctx->loop);
 }
 
+static void
+pk_console_client_finished_cb (GObject *object, GAsyncResult *res, gpointer data)
+{
+	g_autoptr(GError) error = NULL;
+	g_autoptr(PkResults) results = NULL;
+	PkConsoleCtx *ctx = (PkConsoleCtx *) data;
+
+	/* no more progress */
```

Comment:
Couldn't this code move into the generic function as well?

Reply:
There were some issues with the progress bar, though I can't reproduce it now.

Will check and move it to generic function.

Reply:
Good - if it can't be generalized, having it like this would also be okay, but at least in theory, oder of operations between finishing the async callback and ending the progress bar display shouldn't matter...

Reply:
I couldn't reproduce it. So, I'm moving it to generic function.